### PR TITLE
feat(store-sync): recs queries on query-cache

### PR DESCRIPTION
**Pull request by yonadaaa**

### CODE REVIEW

File: packages/store-sync/src/query-cache/common.ts [hunk 1]
```diff
@@ -41,13 +41,13 @@ type queryConditions<tables extends Tables> = {
   [tableName in keyof tables]: tableConditions<tableName & string, tables[tableName]>;
 }[keyof tables];
 
+export type QuerySubjects<tables extends Tables = Tables> = {
+  readonly [k in keyof tables]?: readonly [keyof tables[k]["schema"], ...(keyof tables[k]["schema"])[]];
+};
```
Comment (holic) [inline packages/store-sync/src/query-cache/common.ts:46]: not sure if this is any more readable:
```suggestion
export type QuerySubject<table extends Table = Table> = readonly [keyof table["schema"], ...(keyof table["schema"])[]];

export type QuerySubjects<tables extends Tables = Tables> = {
  readonly [k in keyof tables]?: QuerySubject<tables[k]>;
};
```

File: packages/store-sync/src/query-cache/recs/Query.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+type NotQueryFragment<T extends Table> = {
+  type: QueryFragmentType.Not;
+  table: T;
+};
+
+type HasValueQueryFragment<T extends Table> = {
+  type: QueryFragmentType.HasValue;
+  table: T;
+  value: SchemaToPrimitives<T["valueSchema"]>;
```
Comment (holic) [inline packages/store-sync/src/query-cache/recs/Query.ts:24]: should switch these to `schema`

File: packages/store-sync/src/query-cache/common.ts (hunk 1 above)
Comment (yonadaaa) [reply packages/store-sync/src/query-cache/common.ts:46]: done!

File: packages/store-sync/src/query-cache/recs/Query.ts (hunk 2 above)
Comment (yonadaaa) [reply packages/store-sync/src/query-cache/recs/Query.ts]: interestinggg. really differs from RECS but seems better?

**Comment by holic** [comment]
will be replaced by #2944

**Pull request closed without merging**